// get_stock_data.proto
syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/get_stock_data";

// GetStockServiceというサービスを定義
service GetStockDataService {
  rpc GetStockData (GetStockDataRequest) returns (GetStockDataResponse);
}

// リクエストメッセージを定義
message GetStockDataRequest {
  string ticker = 1;  // 銘柄コードを格納するフィールド
  string period = 2;  // 期間を格納するフィールド
}

// 株価データを持つメッセージを定義
message StockData {
  double open = 1;
  double close = 2;
  double high = 3;
  double low = 4;
  double volume = 5;
}

// レスポンスメッセージを定義
message GetStockDataResponse {
  map<string, StockData> stock_data = 1;  // 期間分の株価データを含むフィールド
}



// python用コンパイルコマンド
// python -m grpc_tools.protoc -I src/get_stock_data --python_out src/get_stock_data --grpc_python_out=src/get_stock_data src/get_stock_data/get_stock_data.proto

// Go用コンパイルコマンド
// protoc --proto_path=../data-analysis-python/src/get_stock_data --go_out=./src/service/ms_gateway/get_stock_data --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_stock_data --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_stock_data/get_stock_data.proto